lib/playlistGenerator/PlaylistGenerator.js

```diff
@@ -233,9 +233,9 @@ var doPersist =  function() {
     that.emit('beforePlaylistChanged', that);
 
     that.isDirty = false;
-    if(that.playlistImp.inner.liveWindowDuration) {
-        that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
-    }
+    that.playlistImp.inner.presentationEndTime = that.playlistImp.getCurrentTime() + that.playlistImp.inner.liveWindowDuration;
+    that.playlistImp.inner.expirationTime = that.playlistImp.inner.presentationEndTime;
```

Comment:
there should be two values:
1. when to return 404
2. when to return end-list

those are different values, in your code it's the same